src/installed.rs

```diff
@@ -181,9 +181,17 @@ impl<C> InstalledFlow<C>
                         Result::Err(Box::new(io::Error::new(io::ErrorKind::UnexpectedEof,
                                                             "couldn't read code")))
                     }
-                    // Remove newline
                     Some(mut code) => {
-                        code.pop();
+                        // Partial backwards compatibilty in case an implementation adds a new line
+                        // due to previous behaviour.
+                        let ends_with_newline = if let Some(ref last_char) = code.chars().last() {
```

Comment:
what about `let code = code.trim_end();`? (https://doc.rust-lang.org/std/string/struct.String.html#method.trim_end)

Reply:
Sure thing if you'd prefer that. I implemented it pretty verbose to stay close to the original behavior and be clear about the intention.
The reason for this code being there is really just for backwards compatibility and could be removed in the next major version upgrade in case this will be merged.
[`code.trim_end_matches(|c| c == '\n')`](https://doc.rust-lang.org/std/string/struct.String.html#method.trim_end_matches) would also be a possibility. We do however need an owned string here and those methods return a `&str` so I would much rather stick with the verbose implementation.

Reply:
Na.. actually I'm sure this check can be done in a one-liner give me a second.

Reply:
ok, sounds fine.